website/blog/2024-10-28-kustomize-vs-helm/index.mdx

```diff
@@ -0,0 +1,179 @@
+
+Kustomize, in contrast, focuses on declarative configurations with plain Kubernetes manifests, making it more intuitive and readable. While Kustomize simplifies configuration management, it lacks some of Helm’s imperative features, like state management, which can be useful in more complex scenarios.
+
+### Kustomize vs Helm:  Ease of use
+
+Helm is often considered easier for packaging entire applications, thanks to its extensive ecosystem of pre-built charts. However, Kustomize isn’t necessarily harder to use, it’s just built for a different purpose. Kustomize shines when making small, targeted modifications to existing resource definition files, making it ideal for managing configurations across multiple environments, rather than packaging whole applications like Helm.
+
+One of Helm’s [drawbacks](/blog/5-helm-shortcomings/) is that its dynamically populated templates can be hard to interpret unless rendered. In contrast, Kustomize manages and modifies plain YAML manifests, making it much easier to read and understand. Engineers are likely to find Kustomize configurations more approachable and intuitive compared to Helm charts when inspecting a repository.
+
+### Kustomize vs Helm: Packaging
+
+Helm is purpose-built for packaging and distributing Kubernetes applications, while Kustomize is more about applying modifications to existing YAML configurations. If you need full application lifecycle management, Helm might be better suited. For granular resource customization, Kustomize is the way to go.
+
+### Native kubectl integration
+
+Kustomize is natively integrated with [kubectl](/guides/kubectl/) from version 1.14 onwards, meaning you can use it directly without needing any additional tools. Helm, on the other hand, requires the Helm CLI.
+
+Install `kubectl` and run:
+```
+kubectl kustomize <kustomization_directory>
+```
+
+Helm on the other hand is custom built to be used in Kubernetes environmnets but it is not integrated into `kubectl`, nor is it considered “kubernetes-native“.
+
+### Declarative vs. Imperative
+
+-   Kustomize (Declarative), focuses purely on defining and applying the desired state with no scripting or state management, keeping things straightforward and predictable.  
+-   Helm (Declarative and Imperative), supports declarative configurations but also allows for imperative operations like running hooks and managing deployment state, adding flexibility at the cost of complexity.
+    
+
+### Mutable vs. Immutable
+
+-   **Helm** manages a mutable state, meaning that it tracks the history of each deployment through it’s handling of [“releases”](/blog/what-is-helm-in-kubernetes/#helm-chart-releases).   
+-   Kustomize works in an immutable way, it doesn't keep track of state. It simply generates the desired state of Kubernetes resources at any given point, so it's up to the user (or other tools) to manage changes over time. 
```

Comment:
I understand what you want to say here, but I don't think mutability is a good term for this. In my opinion, helm release management makes it somewhat "less mutable", if anything. 

I think what lies at the root of what you want to say is something like "history management", in the sense that kustomize doesn't allow you to delete resources removed from the manifest. But it should be noted that something like this can actually be achieved with a combination of the `labels` feature in kustomize and the `--prune` flag when using `kubectl apply`.

Reply:
that's exactly right yeah, that a better way to put it!